src/AzureIoTHubClient.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Arduino. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <time.h>
+#include <sys/time.h>
+
+#include <AzureIoTHubClient.h>
+
+Client* AzureIoTHubClient::sslClient = NULL;
+
+AzureIoTHubClient::AzureIoTHubClient(Client& sslClient)
+{
+    AzureIoTHubClient::sslClient = &sslClient;
+}
+
+void AzureIoTHubClient::begin()
+{
+#if defined(ARDUINO_ARCH_ESP8266)
+    Serial.setDebugOutput(true);
+#endif
+}
+
+void AzureIoTHubClient::setEpochTime(unsigned long epochTime)
```

Comment:
I see. Ok the posix style requirement makes sense. We can keep it like this for now (with setEpochTime). The agenttime implementation is an interesting idea. Lets create an Issue to track that enhancement since some of that might get easier or more complicated as we figure out the upstream sdk code flow process. Currently it would be more unique code here that should come from upstream if we go that route.